runtime/src/utils/handle.rs

```diff
@@ -49,12 +52,13 @@ where
             // Handle result
             let result = match result {
                 Ok(result) => Ok(result),
-                Err(err) => {
-                    if !catch_panic {
-                        resume_unwind(err);
-                    }
-                    let err = extract_panic_message(&*err);
+                Err(panic) => {
+                    let err = extract_panic_message(&*panic);
                     error!(?err, "task panicked");
+                    if let Some(panicker) = panicker.as_ref() {
+                        panicker.notify(panic);
```

Comment:
We should not return from the task in this case, otherwise the result could be processed before the panic?

Reply:
Handled in #1791